src/components/InputField.tsx

```diff
@@ -1,123 +1,105 @@
-  const customInvalidMsg = customBrowserMessages[props.type ?? 'default'] ?? customBrowserMessages.default
+  const [titleId, setTitleId] = useState('')
+  const [titleElement, setTitleElement] = useState<HTMLElement | null>(null)
+
+  const warningLabelText = warningLabels[`${type}`]
+  const inputIsValid = !(title in errors)
 
   useEffect(() => {
     setLabelId(uniqueId.uniqueId('input-label-'))
+    setTitleId(uniqueId.uniqueId('input-title-'))
   }, [])
 
   useEffect(() => {
-    setLabelElement(required ? document.getElementById(labelId) : null)
+    if (labelId) setLabelElement(document.getElementById(labelId))
   }, [labelId])
 
-  const onInput = (e: any, callback = false) => {
-    // supposed to be (e: React.FormEvent<HTMLInputElement>) but TS will highlight (e.target.setCustomValidity) as invalid, that's why (any)
-    if (required && labelElement) {
-      if (!callback) e.target.setCustomValidity('')
-      labelElement.style.opacity = INVISIBLE // hide tooltip
-    }
+  useEffect(() => {
+    if (titleId) setTitleElement(document.getElementById(titleId))
+  }, [titleId])
+
+  /* eslint-disable no-param-reassign */
+  const tooltipFadeIn = () => {
+    if (labelElement) labelElement.style.visibility = 'visible'
+    if (titleElement) titleElement.style.color = primary
+  }
+  /* eslint-disable no-param-reassign */
+  const tooltipFadeOut = () => {
+    if (labelElement) labelElement.style.visibility = 'hidden'
+    if (titleElement) titleElement.style.color = white
   }
 
-  const onInvalid = async (e: any) => {
-    if (required && labelElement) {
-      labelElement.style.opacity = VISIBLE // show tooltip
-      e.target.setCustomValidity(customInvalidMsg) // overrides custom message displayed by browser onInvalid
-      setTimeout(onInput, 2000, e, true) // hide tooltip after 2s
-    }
+  const verifyField = () => {
+    if (inputIsValid) tooltipFadeOut()
+    else tooltipFadeIn()
   }
 
+  /* eslint-disable react/jsx-props-no-spreading */
+  /* eslint-disable object-shorthand */
   return (
     <InputFieldContainer>
-      <TooltipStyled>{required && <Tooltip id={labelId} label={warningLabel} />}</TooltipStyled>
-      <InputFieldStyled>
-        <InputFieldTitle color={white} fontType={h2}>
+      <InputFieldHeader>
+        <InputFieldTitle id={titleId} fontType={h2}>
           {title}
           <span style={{ paddingLeft: '0.1rem' }}>:</span> {/* buffer between title and colon */}
         </InputFieldTitle>
-        <TextInput
-          type={props.type}
-          value={value}
-          pattern={checkPattern}
-          onChange={(e) => setValue(e.target.value)}
-          onInput={onInput}
-          onInvalid={onInvalid}
-          bottomColorActive={white}
-          bottomColorInactive={gray}
-          bottomColorInvalid={danger}
-          fontType={input}
-          required={required}
-        />
-      </InputFieldStyled>
+        <WarningLabel id={labelId} label={warningLabelText} />
```

Comment:
maybe doing the following would be sufficient instead of using id to toggle the visibility? Since react hook form does toggle the error field based on the inputs
```suggestion
        {props.errors && <WarningLabel id={labelId} label={warningLabelText} />}
```

Reply:
changed